docs/DeveloperGuide.md

```diff
@@ -1,34 +1,833 @@
+<br>Throughout the lifespan of the program, various errors may occur. The `ErrorHandling` class stores the various errors that could occur. The expected errors usually stem from invalid user input or Input Output (IO) errors during file loading. The `Ui` class informs the users of the errors detected and suggests actions for rectification. <br>
+<img src = "images/designDG/designSequence.png" width = "700"> <br><br>
+This shows the flow of the program for one user command input.
+
+## 4. Implementation
+
+### AddCommand
+
+The `add` command allows users to add a new show which they are watching to the `ShowList`. It is invoked by the 
+`InputParser` class.
+
+Given below is an example of usage scenario of how the `add` command behaves at each step.
+
+**Step 1**
+
+* The user types in `add friends 2 10,10 30` , adding the `show` to the `Showlist` . The details added include the title
+of the show, the number of seasons of the show, the number of episodes in each season(separated by the comma) and 
+the duration of each episode. 
+
+The `InputParser` class calls the `parseInput` method to parse the command.
+
+> **[NOTE]** 
+>
+> `ArrayOutOfBoundsException` and `NullPointerException` is thrown when the number of arguments entered by the user is incorrect.
+
+**Step 2**
+
+* The `AddCommand` class calls the `AddCommand()` method which then creates a new instance of the `Show` class.
+
+
+**Step 3**
+* The user input is tokenized by the `AddCommand` method into 4 seperate parameters. (Title,Number of seasons,Number of
+episodes for each season respectively,Duration of an episode)
+
+
+**Step 4**
+
+* A new `Show` instance is created with the 4 parameters created in step 3 using the default constructor for show.
+
+**Step 5**
+
+* The `Show` is added to the `ShowList`.
+
+
+**Step 6** 
+
+* The changes will be reflected to the user. Concurrently, the changes will be saved into the showList.txt file.
+
+### EditCommand
+
+The `edit` command allows the user to change the details of each show that they are watching after they have added the
+show. It is self-contained, it includes its own parser and methods which prompts the user to change any parameter they 
+wish, after the user enters `done`, the `edit` command replaces the old entry with the updated one.
+
+Given below is an example of usage scenario of how the edit command behaves at each step and a sequence diagram to
+illustrate the steps in a visual form.
+
+> **[NOTE]** 
+>
+> `NullPointerException` will be thrown when show entered by user is not found in the showlist.
+
+<img src = "images/EditCommand.png" width = "700">
```

Comment:
apart from the comments given by @jlifah, I think the life line of the user is missing from the diagram
![image](https://user-images.githubusercontent.com/57038559/97525985-22613a80-19e3-11eb-904d-26af607203e1.png)